README.md
# client

The Jira Cloud platform REST API
- API version: 1001.0.0-SNAPSHOT
  - Build date: 2021-03-10T13:09:14.296+05:30[Asia/Kolkata]

Jira Cloud platform REST API documentation


*Automatically generated by the [Swagger Codegen](https://github.com/swagger-api/swagger-codegen)*


## Requirements

Building the API client library requires:
1. Java 1.7+
2. Maven/Gradle

## Installation

To install the API client library to your local Maven repository, simply execute:

```shell
mvn clean install
```

To deploy it to a remote Maven repository instead, configure the settings of the repository and execute:

```shell
mvn clean deploy
```

Refer to the [OSSRH Guide](http://central.sonatype.org/pages/ossrh-guide.html) for more information.

### Maven users

Add this dependency to your project's POM:

```xml
<dependency>
  <groupId>com.xminds.jira.rest</groupId>
  <artifactId>client</artifactId>
  <version>1.0.0</version>
  <scope>compile</scope>
</dependency>
```

### Gradle users

Add this dependency to your project's build file:

```groovy
compile "com.xminds.jira.rest:client:1.0.0"
```

### Others

At first generate the JAR by executing:

```shell
mvn clean package
```

Then manually install the following JARs:

* `target/client-1.0.0.jar`
* `target/lib/*.jar`

## Getting Started

Please follow the [installation](#installation) instruction and execute the following Java code:

```java
import com.xminds.jira.rest.client.*;
import com.xminds.jira.rest.client.auth.*;
import com.xminds.jira.rest.client.model.*;
import com.xminds.jira.rest.client.api.AppPropertiesApi;

import java.io.File;
import java.util.*;

public class AppPropertiesApiExample {

    public static void main(String[] args) {
        
        AppPropertiesApi apiInstance = new AppPropertiesApi();
        String addonKey = "addonKey_example"; // String | The key of the app, as defined in its descriptor.
        String propertyKey = "propertyKey_example"; // String | The key of the property.
        try {
            apiInstance.addonPropertiesResourceDeleteAddonPropertyDelete(addonKey, propertyKey);
        } catch (ApiException e) {
            System.err.println("Exception when calling AppPropertiesApi#addonPropertiesResourceDeleteAddonPropertyDelete");
            e.printStackTrace();
        }
    }
}
import com.xminds.jira.rest.client.*;
import com.xminds.jira.rest.client.auth.*;
import com.xminds.jira.rest.client.model.*;
import com.xminds.jira.rest.client.api.AppPropertiesApi;

import java.io.File;
import java.util.*;

public class AppPropertiesApiExample {

    public static void main(String[] args) {
        
        AppPropertiesApi apiInstance = new AppPropertiesApi();
        String addonKey = "addonKey_example"; // String | The key of the app, as defined in its descriptor.
        try {
            PropertyKeys result = apiInstance.addonPropertiesResourceGetAddonPropertiesGet(addonKey);
            System.out.println(result);
        } catch (ApiException e) {
            System.err.println("Exception when calling AppPropertiesApi#addonPropertiesResourceGetAddonPropertiesGet");
            e.printStackTrace();
        }
    }
}
import com.xminds.jira.rest.client.*;
import com.xminds.jira.rest.client.auth.*;
import com.xminds.jira.rest.client.model.*;
import com.xminds.jira.rest.client.api.AppPropertiesApi;

import java.io.File;
import java.util.*;

public class AppPropertiesApiExample {

    public static void main(String[] args) {
        
        AppPropertiesApi apiInstance = new AppPropertiesApi();
        String addonKey = "addonKey_example"; // String | The key of the app, as defined in its descriptor.
        String propertyKey = "propertyKey_example"; // String | The key of the property.
        try {
            EntityProperty result = apiInstance.addonPropertiesResourceGetAddonPropertyGet(addonKey, propertyKey);
            System.out.println(result);
        } catch (ApiException e) {
            System.err.println("Exception when calling AppPropertiesApi#addonPropertiesResourceGetAddonPropertyGet");
            e.printStackTrace();
        }
    }
}
import com.xminds.jira.rest.client.*;
import com.xminds.jira.rest.client.auth.*;
import com.xminds.jira.rest.client.model.*;
import com.xminds.jira.rest.client.api.AppPropertiesApi;

import java.io.File;
import java.util.*;

public class AppPropertiesApiExample {

    public static void main(String[] args) {
        
        AppPropertiesApi apiInstance = new AppPropertiesApi();
        Object body = null; // Object | 
        String addonKey = "addonKey_example"; // String | The key of the app, as defined in its descriptor.
        String propertyKey = "propertyKey_example"; // String | The key of the property.
        try {
            OperationMessage result = apiInstance.addonPropertiesResourcePutAddonPropertyPut(body, addonKey, propertyKey);
            System.out.println(result);
        } catch (ApiException e) {
            System.err.println("Exception when calling AppPropertiesApi#addonPropertiesResourcePutAddonPropertyPut");
            e.printStackTrace();
        }
    }
}
```

## Documentation for API Endpoints

All URIs are relative to *https://your-domain.atlassian.com*

Class | Method | HTTP request | Description
------------ | ------------- | ------------- | -------------
*AppPropertiesApi* | [**addonPropertiesResourceDeleteAddonPropertyDelete**](docs/AppPropertiesApi.md#addonPropertiesResourceDeleteAddonPropertyDelete) | **DELETE** /rest/atlassian-connect/1/addons/{addonKey}/properties/{propertyKey} | Delete app property
*AppPropertiesApi* | [**addonPropertiesResourceGetAddonPropertiesGet**](docs/AppPropertiesApi.md#addonPropertiesResourceGetAddonPropertiesGet) | **GET** /rest/atlassian-connect/1/addons/{addonKey}/properties | Get app properties
*AppPropertiesApi* | [**addonPropertiesResourceGetAddonPropertyGet**](docs/AppPropertiesApi.md#addonPropertiesResourceGetAddonPropertyGet) | **GET** /rest/atlassian-connect/1/addons/{addonKey}/properties/{propertyKey} | Get app property
*AppPropertiesApi* | [**addonPropertiesResourcePutAddonPropertyPut**](docs/AppPropertiesApi.md#addonPropertiesResourcePutAddonPropertyPut) | **PUT** /rest/atlassian-connect/1/addons/{addonKey}/properties/{propertyKey} | Set app property
*ApplicationRolesApi* | [**getAllApplicationRoles**](docs/ApplicationRolesApi.md#getAllApplicationRoles) | **GET** /rest/api/3/applicationrole | Get all application roles
*ApplicationRolesApi* | [**getApplicationRole**](docs/ApplicationRolesApi.md#getApplicationRole) | **GET** /rest/api/3/applicationrole/{key} | Get application role
*AuditRecordsApi* | [**getAuditRecords**](docs/AuditRecordsApi.md#getAuditRecords) | **GET** /rest/api/3/auditing/record | Get audit records
*AvatarsApi* | [**deleteAvatar**](docs/AvatarsApi.md#deleteAvatar) | **DELETE** /rest/api/3/universal_avatar/type/{type}/owner/{owningObjectId}/avatar/{id} | Delete avatar
*AvatarsApi* | [**getAllSystemAvatars**](docs/AvatarsApi.md#getAllSystemAvatars) | **GET** /rest/api/3/avatar/{type}/system | Get system avatars by type
*AvatarsApi* | [**getAvatars**](docs/AvatarsApi.md#getAvatars) | **GET** /rest/api/3/universal_avatar/type/{type}/owner/{entityId} | Get avatars
*AvatarsApi* | [**storeAvatar**](docs/AvatarsApi.md#storeAvatar) | **POST** /rest/api/3/universal_avatar/type/{type}/owner/{entityId} | Load avatar
*DashboardsApi* | [**copyDashboard**](docs/DashboardsApi.md#copyDashboard) | **POST** /rest/api/3/dashboard/{id}/copy | Copy dashboard
*DashboardsApi* | [**createDashboard**](docs/DashboardsApi.md#createDashboard) | **POST** /rest/api/3/dashboard | Create dashboard
*DashboardsApi* | [**deleteDashboard**](docs/DashboardsApi.md#deleteDashboard) | **DELETE** /rest/api/3/dashboard/{id} | Delete dashboard
*DashboardsApi* | [**deleteDashboardItemProperty**](docs/DashboardsApi.md#deleteDashboardItemProperty) | **DELETE** /rest/api/3/dashboard/{dashboardId}/items/{itemId}/properties/{propertyKey} | Delete dashboard item property
*DashboardsApi* | [**getAllDashboards**](docs/DashboardsApi.md#getAllDashboards) | **GET** /rest/api/3/dashboard | Get all dashboards
*DashboardsApi* | [**getDashboard**](docs/DashboardsApi.md#getDashboard) | **GET** /rest/api/3/dashboard/{id} | Get dashboard
*DashboardsApi* | [**getDashboardItemProperty**](docs/DashboardsApi.md#getDashboardItemProperty) | **GET** /rest/api/3/dashboard/{dashboardId}/items/{itemId}/properties/{propertyKey} | Get dashboard item property
*DashboardsApi* | [**getDashboardItemPropertyKeys**](docs/DashboardsApi.md#getDashboardItemPropertyKeys) | **GET** /rest/api/3/dashboard/{dashboardId}/items/{itemId}/properties | Get dashboard item property keys
*DashboardsApi* | [**getDashboardsPaginated**](docs/DashboardsApi.md#getDashboardsPaginated) | **GET** /rest/api/3/dashboard/search | Search for dashboards
*DashboardsApi* | [**setDashboardItemProperty**](docs/DashboardsApi.md#setDashboardItemProperty) | **PUT** /rest/api/3/dashboard/{dashboardId}/items/{itemId}/properties/{propertyKey} | Set dashboard item property
*DashboardsApi* | [**updateDashboard**](docs/DashboardsApi.md#updateDashboard) | **PUT** /rest/api/3/dashboard/{id} | Update dashboard
*DynamicModulesApi* | [**dynamicModulesResourceGetModulesGet**](docs/DynamicModulesApi.md#dynamicModulesResourceGetModulesGet) | **GET** /rest/atlassian-connect/1/app/module/dynamic | Get modules
*DynamicModulesApi* | [**dynamicModulesResourceRegisterModulesPost**](docs/DynamicModulesApi.md#dynamicModulesResourceRegisterModulesPost) | **POST** /rest/atlassian-connect/1/app/module/dynamic | Register modules
*DynamicModulesApi* | [**dynamicModulesResourceRemoveModulesDelete**](docs/DynamicModulesApi.md#dynamicModulesResourceRemoveModulesDelete) | **DELETE** /rest/atlassian-connect/1/app/module/dynamic | Remove modules
*FilterSharingApi* | [**addSharePermission**](docs/FilterSharingApi.md#addSharePermission) | **POST** /rest/api/3/filter/{id}/permission | Add share permission
*FilterSharingApi* | [**deleteSharePermission**](docs/FilterSharingApi.md#deleteSharePermission) | **DELETE** /rest/api/3/filter/{id}/permission/{permissionId} | Delete share permission
*FilterSharingApi* | [**getDefaultShareScope**](docs/FilterSharingApi.md#getDefaultShareScope) | **GET** /rest/api/3/filter/defaultShareScope | Get default share scope
*FilterSharingApi* | [**getSharePermission**](docs/FilterSharingApi.md#getSharePermission) | **GET** /rest/api/3/filter/{id}/permission/{permissionId} | Get share permission
*FilterSharingApi* | [**getSharePermissions**](docs/FilterSharingApi.md#getSharePermissions) | **GET** /rest/api/3/filter/{id}/permission | Get share permissions
*FilterSharingApi* | [**setDefaultShareScope**](docs/FilterSharingApi.md#setDefaultShareScope) | **PUT** /rest/api/3/filter/defaultShareScope | Set default share scope
*FiltersApi* | [**createFilter**](docs/FiltersApi.md#createFilter) | **POST** /rest/api/3/filter | Create filter
*FiltersApi* | [**deleteFavouriteForFilter**](docs/FiltersApi.md#deleteFavouriteForFilter) | **DELETE** /rest/api/3/filter/{id}/favourite | Remove filter as favorite
*FiltersApi* | [**deleteFilter**](docs/FiltersApi.md#deleteFilter) | **DELETE** /rest/api/3/filter/{id} | Delete filter
*FiltersApi* | [**getColumns**](docs/FiltersApi.md#getColumns) | **GET** /rest/api/3/filter/{id}/columns | Get columns
*FiltersApi* | [**getFavouriteFilters**](docs/FiltersApi.md#getFavouriteFilters) | **GET** /rest/api/3/filter/favourite | Get favorite filters
*FiltersApi* | [**getFilter**](docs/FiltersApi.md#getFilter) | **GET** /rest/api/3/filter/{id} | Get filter
*FiltersApi* | [**getFilters**](docs/FiltersApi.md#getFilters) | **GET** /rest/api/3/filter | Get filters
*FiltersApi* | [**getFiltersPaginated**](docs/FiltersApi.md#getFiltersPaginated) | **GET** /rest/api/3/filter/search | Search for filters
*FiltersApi* | [**getMyFilters**](docs/FiltersApi.md#getMyFilters) | **GET** /rest/api/3/filter/my | Get my filters
*FiltersApi* | [**resetColumns**](docs/FiltersApi.md#resetColumns) | **DELETE** /rest/api/3/filter/{id}/columns | Reset columns
*FiltersApi* | [**setColumns**](docs/FiltersApi.md#setColumns) | **PUT** /rest/api/3/filter/{id}/columns | Set columns
*FiltersApi* | [**setFavouriteForFilter**](docs/FiltersApi.md#setFavouriteForFilter) | **PUT** /rest/api/3/filter/{id}/favourite | Add filter as favorite
*FiltersApi* | [**updateFilter**](docs/FiltersApi.md#updateFilter) | **PUT** /rest/api/3/filter/{id} | Update filter
*GroupAndUserPickerApi* | [**findUsersAndGroups**](docs/GroupAndUserPickerApi.md#findUsersAndGroups) | **GET** /rest/api/3/groupuserpicker | Find users and groups
*GroupsApi* | [**addUserToGroup**](docs/GroupsApi.md#addUserToGroup) | **POST** /rest/api/3/group/user | Add user to group
*GroupsApi* | [**bulkGetGroups**](docs/GroupsApi.md#bulkGetGroups) | **GET** /rest/api/3/group/bulk | Bulk get groups
*GroupsApi* | [**createGroup**](docs/GroupsApi.md#createGroup) | **POST** /rest/api/3/group | Create group
*GroupsApi* | [**findGroups**](docs/GroupsApi.md#findGroups) | **GET** /rest/api/3/groups/picker | Find groups
*GroupsApi* | [**getGroup**](docs/GroupsApi.md#getGroup) | **GET** /rest/api/3/group | Get group
*GroupsApi* | [**getUsersFromGroup**](docs/GroupsApi.md#getUsersFromGroup) | **GET** /rest/api/3/group/member | Get users from group
*GroupsApi* | [**removeGroup**](docs/GroupsApi.md#removeGroup) | **DELETE** /rest/api/3/group | Remove group
*GroupsApi* | [**removeUserFromGroup**](docs/GroupsApi.md#removeUserFromGroup) | **DELETE** /rest/api/3/group/user | Remove user from group
*IssueAttachmentsApi* | [**addAttachment**](docs/IssueAttachmentsApi.md#addAttachment) | **POST** /rest/api/3/issue/{issueIdOrKey}/attachments | Add attachment
*IssueAttachmentsApi* | [**expandAttachmentForHumans**](docs/IssueAttachmentsApi.md#expandAttachmentForHumans) | **GET** /rest/api/3/attachment/{id}/expand/human | Get all metadata for an expanded attachment
*IssueAttachmentsApi* | [**expandAttachmentForMachines**](docs/IssueAttachmentsApi.md#expandAttachmentForMachines) | **GET** /rest/api/3/attachment/{id}/expand/raw | Get contents metadata for an expanded attachment
*IssueAttachmentsApi* | [**getAttachment**](docs/IssueAttachmentsApi.md#getAttachment) | **GET** /rest/api/3/attachment/{id} | Get attachment metadata
*IssueAttachmentsApi* | [**getAttachmentMeta**](docs/IssueAttachmentsApi.md#getAttachmentMeta) | **GET** /rest/api/3/attachment/meta | Get Jira attachment settings
*IssueAttachmentsApi* | [**removeAttachment**](docs/IssueAttachmentsApi.md#removeAttachment) | **DELETE** /rest/api/3/attachment/{id} | Delete attachment
*IssueCommentPropertiesApi* | [**deleteCommentProperty**](docs/IssueCommentPropertiesApi.md#deleteCommentProperty) | **DELETE** /rest/api/3/comment/{commentId}/properties/{propertyKey} | Delete comment property
*IssueCommentPropertiesApi* | [**getCommentProperty**](docs/IssueCommentPropertiesApi.md#getCommentProperty) | **GET** /rest/api/3/comment/{commentId}/properties/{propertyKey} | Get comment property
*IssueCommentPropertiesApi* | [**getCommentPropertyKeys**](docs/IssueCommentPropertiesApi.md#getCommentPropertyKeys) | **GET** /rest/api/3/comment/{commentId}/properties | Get comment property keys
*IssueCommentPropertiesApi* | [**setCommentProperty**](docs/IssueCommentPropertiesApi.md#setCommentProperty) | **PUT** /rest/api/3/comment/{commentId}/properties/{propertyKey} | Set comment property
*IssueCommentsApi* | [**addComment**](docs/IssueCommentsApi.md#addComment) | **POST** /rest/api/3/issue/{issueIdOrKey}/comment | Add comment
*IssueCommentsApi* | [**deleteComment**](docs/IssueCommentsApi.md#deleteComment) | **DELETE** /rest/api/3/issue/{issueIdOrKey}/comment/{id} | Delete comment
*IssueCommentsApi* | [**getComment**](docs/IssueCommentsApi.md#getComment) | **GET** /rest/api/3/issue/{issueIdOrKey}/comment/{id} | Get comment
*IssueCommentsApi* | [**getComments**](docs/IssueCommentsApi.md#getComments) | **GET** /rest/api/3/issue/{issueIdOrKey}/comment | Get comments
*IssueCommentsApi* | [**getCommentsByIds**](docs/IssueCommentsApi.md#getCommentsByIds) | **POST** /rest/api/3/comment/list | Get comments by IDs
*IssueCommentsApi* | [**updateComment**](docs/IssueCommentsApi.md#updateComment) | **PUT** /rest/api/3/issue/{issueIdOrKey}/comment/{id} | Update comment
*IssueCustomFieldContextsApi* | [**addIssueTypesToContext**](docs/IssueCustomFieldContextsApi.md#addIssueTypesToContext) | **PUT** /rest/api/3/field/{fieldId}/context/{contextId}/issuetype | Add issue types to context
*IssueCustomFieldContextsApi* | [**assignProjectsToCustomFieldContext**](docs/IssueCustomFieldContextsApi.md#assignProjectsToCustomFieldContext) | **PUT** /rest/api/3/field/{fieldId}/context/{contextId}/project | Assign custom field context to projects
*IssueCustomFieldContextsApi* | [**createCustomFieldContext**](docs/IssueCustomFieldContextsApi.md#createCustomFieldContext) | **POST** /rest/api/3/field/{fieldId}/context | Create custom field context
*IssueCustomFieldContextsApi* | [**deleteCustomFieldContext**](docs/IssueCustomFieldContextsApi.md#deleteCustomFieldContext) | **DELETE** /rest/api/3/field/{fieldId}/context/{contextId} | Delete custom field context
*IssueCustomFieldContextsApi* | [**getContextsForField**](docs/IssueCustomFieldContextsApi.md#getContextsForField) | **GET** /rest/api/3/field/{fieldId}/context | Get custom field contexts
*IssueCustomFieldContextsApi* | [**getCustomFieldContextsForProjectsAndIssueTypes**](docs/IssueCustomFieldContextsApi.md#getCustomFieldContextsForProjectsAndIssueTypes) | **POST** /rest/api/3/field/{fieldId}/context/mapping | Get custom field contexts for projects and issue types
*IssueCustomFieldContextsApi* | [**getDefaultValues**](docs/IssueCustomFieldContextsApi.md#getDefaultValues) | **GET** /rest/api/3/field/{fieldId}/context/defaultValue | Get custom field contexts default values
*IssueCustomFieldContextsApi* | [**getIssueTypeMappingsForContexts**](docs/IssueCustomFieldContextsApi.md#getIssueTypeMappingsForContexts) | **GET** /rest/api/3/field/{fieldId}/context/issuetypemapping | Get issue types for custom field context
*IssueCustomFieldContextsApi* | [**getProjectContextMapping**](docs/IssueCustomFieldContextsApi.md#getProjectContextMapping) | **GET** /rest/api/3/field/{fieldId}/context/projectmapping | Get project mappings for custom field context
*IssueCustomFieldContextsApi* | [**removeCustomFieldContextFromProjects**](docs/IssueCustomFieldContextsApi.md#removeCustomFieldContextFromProjects) | **POST** /rest/api/3/field/{fieldId}/context/{contextId}/project/remove | Remove custom field context from projects
*IssueCustomFieldContextsApi* | [**removeIssueTypesFromContext**](docs/IssueCustomFieldContextsApi.md#removeIssueTypesFromContext) | **POST** /rest/api/3/field/{fieldId}/context/{contextId}/issuetype/remove | Remove issue types from context
*IssueCustomFieldContextsApi* | [**setDefaultValues**](docs/IssueCustomFieldContextsApi.md#setDefaultValues) | **PUT** /rest/api/3/field/{fieldId}/context/defaultValue | Set custom field contexts default values
*IssueCustomFieldContextsApi* | [**updateCustomFieldContext**](docs/IssueCustomFieldContextsApi.md#updateCustomFieldContext) | **PUT** /rest/api/3/field/{fieldId}/context/{contextId} | Update custom field context
*IssueCustomFieldOptionsApi* | [**createCustomFieldOption**](docs/IssueCustomFieldOptionsApi.md#createCustomFieldOption) | **POST** /rest/api/3/field/{fieldId}/context/{contextId}/option | Create custom field options (context)
*IssueCustomFieldOptionsApi* | [**createCustomFieldOptions**](docs/IssueCustomFieldOptionsApi.md#createCustomFieldOptions) | **POST** /rest/api/3/customField/{fieldId}/option | Create custom field options
*IssueCustomFieldOptionsApi* | [**deleteCustomFieldOption**](docs/IssueCustomFieldOptionsApi.md#deleteCustomFieldOption) | **DELETE** /rest/api/3/field/{fieldId}/context/{contextId}/option/{optionId} | Delete custom field options (context)
*IssueCustomFieldOptionsApi* | [**getCustomFieldOption**](docs/IssueCustomFieldOptionsApi.md#getCustomFieldOption) | **GET** /rest/api/3/customFieldOption/{id} | Get custom field option
*IssueCustomFieldOptionsApi* | [**getOptionsForContext**](docs/IssueCustomFieldOptionsApi.md#getOptionsForContext) | **GET** /rest/api/3/field/{fieldId}/context/{contextId}/option | Get custom field options (context)
*IssueCustomFieldOptionsApi* | [**getOptionsForField**](docs/IssueCustomFieldOptionsApi.md#getOptionsForField) | **GET** /rest/api/3/customField/{fieldId}/option | Get options for field
*IssueCustomFieldOptionsApi* | [**reorderCustomFieldOptions**](docs/IssueCustomFieldOptionsApi.md#reorderCustomFieldOptions) | **PUT** /rest/api/3/field/{fieldId}/context/{contextId}/option/move | Reorder custom field options (context)
*IssueCustomFieldOptionsApi* | [**updateCustomFieldOption**](docs/IssueCustomFieldOptionsApi.md#updateCustomFieldOption) | **PUT** /rest/api/3/field/{fieldId}/context/{contextId}/option | Update custom field options (context)
*IssueCustomFieldOptionsApi* | [**updateCustomFieldOptions**](docs/IssueCustomFieldOptionsApi.md#updateCustomFieldOptions) | **PUT** /rest/api/3/customField/{fieldId}/option | Update custom field options
*IssueCustomFieldOptionsAppsApi* | [**createIssueFieldOption**](docs/IssueCustomFieldOptionsAppsApi.md#createIssueFieldOption) | **POST** /rest/api/3/field/{fieldKey}/option | Create issue field option

docs/Changelog.md
# Changelog

## Properties
Name | Type | Description | Notes
------------ | ------------- | ------------- | -------------
**id** | **String** | The ID of the changelog. |  [optional]
**author** | **AllOfChangelogAuthor** | The user who made the change. |  [optional]
**created** | [**OffsetDateTime**](OffsetDateTime.md) | The date on which the change took place. |  [optional]
**items** | [**List&lt;ChangeDetails&gt;**](ChangeDetails.md) | The list of items changed. |  [optional]
**historyMetadata** | **AllOfChangelogHistoryMetadata** | The history metadata associated with the changed. |  [optional]


src/main/java/com/xminds/jira/rest/client/api/JqlApi.java
package com.xminds.jira.rest.client.api;

import com.xminds.jira.rest.client.ApiException;
import com.xminds.jira.rest.client.ApiClient;
import com.xminds.jira.rest.client.Configuration;
import com.xminds.jira.rest.client.Pair;

import javax.ws.rs.core.GenericType;

import com.xminds.jira.rest.client.model.AutoCompleteSuggestions;
import com.xminds.jira.rest.client.model.ConvertedJQLQueries;
import com.xminds.jira.rest.client.model.ErrorCollection;
import com.xminds.jira.rest.client.model.JQLPersonalDataMigrationRequest;
import com.xminds.jira.rest.client.model.JQLReferenceData;
import com.xminds.jira.rest.client.model.JqlQueriesToParse;
import com.xminds.jira.rest.client.model.ParsedJqlQueries;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.JavaClientCodegen", date = "2021-03-10T13:09:14.296+05:30[Asia/Kolkata]")public class JqlApi {
  private ApiClient apiClient;

  public JqlApi() {
    this(Configuration.getDefaultApiClient());
  }

  public JqlApi(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  public ApiClient getApiClient() {
    return apiClient;
  }

  public void setApiClient(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  /**
   * Get field reference data
   * Returns reference data for JQL searches. This is a downloadable version of the documentation provided in [Advanced searching - fields reference](https://confluence.atlassian.com/x/gwORLQ) and [Advanced searching - functions reference](https://confluence.atlassian.com/x/hgORLQ), along with a list of JQL-reserved words. Use this information to assist with the programmatic creation of JQL queries or the validation of queries built in a custom query builder.  This operation can be accessed anonymously.  **[Permissions](#permissions) required:** None.
   * @return JQLReferenceData
   * @throws ApiException if fails to make API call
   */
  public JQLReferenceData getAutoComplete() throws ApiException {
    Object localVarPostBody = null;
    // create path and map variables
    String localVarPath = "/rest/api/3/jql/autocompletedata";

    // query params
    List<Pair> localVarQueryParams = new ArrayList<Pair>();
    Map<String, String> localVarHeaderParams = new HashMap<String, String>();
    Map<String, Object> localVarFormParams = new HashMap<String, Object>();



    final String[] localVarAccepts = {
      "application/json"
    };
    final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);

    final String[] localVarContentTypes = {
      
    };
    final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);

    String[] localVarAuthNames = new String[] { "OAuth2", "basicAuth" };

    GenericType<JQLReferenceData> localVarReturnType = new GenericType<JQLReferenceData>() {};
    return apiClient.invokeAPI(localVarPath, "GET", localVarQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarAccept, localVarContentType, localVarAuthNames, localVarReturnType);
  }
  /**
   * Get field auto complete suggestions
   * Returns the JQL search auto complete suggestions for a field.  Suggestions can be obtained by providing:   *  &#x60;fieldName&#x60; to get a list of all values for the field.  *  &#x60;fieldName&#x60; and &#x60;fieldValue&#x60; to get a list of values containing the text in &#x60;fieldValue&#x60;.  *  &#x60;fieldName&#x60; and &#x60;predicateName&#x60; to get a list of all predicate values for the field.  *  &#x60;fieldName&#x60;, &#x60;predicateName&#x60;, and &#x60;predicateValue&#x60; to get a list of predicate values containing the text in &#x60;predicateValue&#x60;.  This operation can be accessed anonymously.  **[Permissions](#permissions) required:** None.
   * @param fieldName The name of the field. (optional)
   * @param fieldValue The partial field item name entered by the user. (optional)
   * @param predicateName The name of the [ CHANGED operator predicate](https://confluence.atlassian.com/x/hQORLQ#Advancedsearching-operatorsreference-CHANGEDCHANGED) for which the suggestions are generated. The valid predicate operators are *by*, *from*, and *to*. (optional)
   * @param predicateValue The partial predicate item name entered by the user. (optional)
   * @return AutoCompleteSuggestions
   * @throws ApiException if fails to make API call
   */
  public AutoCompleteSuggestions getFieldAutoCompleteForQueryString(String fieldName, String fieldValue, String predicateName, String predicateValue) throws ApiException {
    Object localVarPostBody = null;
    // create path and map variables
    String localVarPath = "/rest/api/3/jql/autocompletedata/suggestions";

    // query params
    List<Pair> localVarQueryParams = new ArrayList<Pair>();
    Map<String, String> localVarHeaderParams = new HashMap<String, String>();
    Map<String, Object> localVarFormParams = new HashMap<String, Object>();

    localVarQueryParams.addAll(apiClient.parameterToPairs("", "fieldName", fieldName));
    localVarQueryParams.addAll(apiClient.parameterToPairs("", "fieldValue", fieldValue));
    localVarQueryParams.addAll(apiClient.parameterToPairs("", "predicateName", predicateName));
    localVarQueryParams.addAll(apiClient.parameterToPairs("", "predicateValue", predicateValue));


    final String[] localVarAccepts = {
      "application/json"
    };
    final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);

    final String[] localVarContentTypes = {
      
    };
    final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);

    String[] localVarAuthNames = new String[] { "OAuth2", "basicAuth" };

    GenericType<AutoCompleteSuggestions> localVarReturnType = new GenericType<AutoCompleteSuggestions>() {};
    return apiClient.invokeAPI(localVarPath, "GET", localVarQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarAccept, localVarContentType, localVarAuthNames, localVarReturnType);
  }
  /**
   * Convert user identifiers to account IDs in JQL queries
   * Converts one or more JQL queries with user identifiers (username or user key) to equivalent JQL queries with account IDs.  You may wish to use this operation if your system stores JQL queries and you want to make them GDPR-compliant. For more information about GDPR-related changes, see the [migration guide](https://developer.atlassian.com/cloud/jira/platform/deprecation-notice-user-privacy-api-migration-guide/).  **[Permissions](#permissions) required:** Permission to access Jira.
   * @param body  (required)
   * @return ConvertedJQLQueries
   * @throws ApiException if fails to make API call
   */
  public ConvertedJQLQueries migrateQueries(JQLPersonalDataMigrationRequest body) throws ApiException {
    Object localVarPostBody = body;
    // verify the required parameter 'body' is set
    if (body == null) {
      throw new ApiException(400, "Missing the required parameter 'body' when calling migrateQueries");
    }
    // create path and map variables
    String localVarPath = "/rest/api/3/jql/pdcleaner";

    // query params
    List<Pair> localVarQueryParams = new ArrayList<Pair>();
    Map<String, String> localVarHeaderParams = new HashMap<String, String>();
    Map<String, Object> localVarFormParams = new HashMap<String, Object>();



    final String[] localVarAccepts = {
      "application/json"
    };
    final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);

    final String[] localVarContentTypes = {
      "application/json"
    };
    final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);

    String[] localVarAuthNames = new String[] { "OAuth2", "basicAuth" };

    GenericType<ConvertedJQLQueries> localVarReturnType = new GenericType<ConvertedJQLQueries>() {};
    return apiClient.invokeAPI(localVarPath, "POST", localVarQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarAccept, localVarContentType, localVarAuthNames, localVarReturnType);
  }
  /**
   * Parse JQL query
   * Parses and validates JQL queries.  Validation is performed in context of the current user.  This operation can be accessed anonymously.  **[Permissions](#permissions) required:** None.
   * @param body  (required)
   * @param validation How to validate the JQL query and treat the validation results. Validation options include:   *  &#x60;strict&#x60; Returns all errors. If validation fails, the query structure is not returned.  *  &#x60;warn&#x60; Returns all errors. If validation fails but the JQL query is correctly formed, the query structure is returned.  *  &#x60;none&#x60; No validation is performed. If JQL query is correctly formed, the query structure is returned. (optional, default to strict)
   * @return ParsedJqlQueries
   * @throws ApiException if fails to make API call
   */
  public ParsedJqlQueries parseJqlQueries(JqlQueriesToParse body, String validation) throws ApiException {
    Object localVarPostBody = body;
    // verify the required parameter 'body' is set
    if (body == null) {
      throw new ApiException(400, "Missing the required parameter 'body' when calling parseJqlQueries");
    }
    // create path and map variables
    String localVarPath = "/rest/api/3/jql/parse";

    // query params
    List<Pair> localVarQueryParams = new ArrayList<Pair>();
    Map<String, String> localVarHeaderParams = new HashMap<String, String>();
    Map<String, Object> localVarFormParams = new HashMap<String, Object>();

    localVarQueryParams.addAll(apiClient.parameterToPairs("", "validation", validation));


    final String[] localVarAccepts = {
      "application/json"
    };
    final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);

    final String[] localVarContentTypes = {
      "application/json"
    };
    final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);

    String[] localVarAuthNames = new String[] { "OAuth2", "basicAuth" };

    GenericType<ParsedJqlQueries> localVarReturnType = new GenericType<ParsedJqlQueries>() {};
    return apiClient.invokeAPI(localVarPath, "POST", localVarQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarAccept, localVarContentType, localVarAuthNames, localVarReturnType);
  }
}
